pkg/config/sync.go

```diff
@@ -49,6 +51,40 @@ func GetGlobalConfig() *controller.OperatorConfiguration {
 	return internalConfig.DeepCopy()
 }
 
+// TODO: Improve this function description?
+// Checks for an external DevWorkspace-Operator configuration specified by the optional workspace attribute `controller.devfile.io/devworkspace-config`
+// If the `controller.devfile.io/devworkspace-config` attribute is not set, the internal/global DWOC is returned
+// If the `controller.devfile.io/devworkspace-config` attribute is incorrectly set, the internal/global DWOC and an error are returned.
+// If the external DWOC is found on the cluster, it is merged with the internal/global DWOC and returned.
+// If the external DWOC is not found on the cluster, the internal/global DWOC and an error are returned.
+func ResolveConfigForWorkspace(workspace *dw.DevWorkspace, client crclient.Client) (*controller.OperatorConfiguration, error) {
+	internalConfigCopy := internalConfig.DeepCopy()
+	if workspace.Spec.Template.Attributes.Exists(constants.ExternalDevWorkspaceConfiguration) {
+		namespacedName := types.NamespacedName{}
+
+		err := workspace.Spec.Template.Attributes.GetInto(constants.ExternalDevWorkspaceConfiguration, &namespacedName)
+		if err != nil {
+			return internalConfigCopy, fmt.Errorf("failed to read attribute %s in DevWorkspace attributes: %w", constants.ExternalDevWorkspaceConfiguration, err)
```

Comment:
Should return `nil, [error]` (here and elsewhere in the function to force the error to be addressed (and not hidden/ignored). Then, call sites should be something like 
```go
	config, err := wkspConfig.ResolveConfigForWorkspace(rawWorkspace, clusterAPI.Client)
	if err != nil {
		reqLogger.Error(err, "Error applying external DevWorkspace-Operator configuration")
		config = wkspConfig.GetGlobalConfig()
	}
```
i.e. _explicitly_ use the global configuration if we fail to read a custom one.

Reply:
I didn't want to return an error in the case that the attribute is not set in the workspace (cause then it would get logged constantly) so I added another function [`ExternalConfigSet`](https://github.com/devfile/devworkspace-operator/pull/918/commits/5d11520216ec35a96bb1a57d0be6041b287c4528#diff-44d82704651db22ff5ebfa1b636e76edf8e97a075c3cba2ea1714be914d6af74R55) to check before trying to resolve the workspace config. However, I dislike the [if/else case](https://github.com/devfile/devworkspace-operator/pull/918/commits/5d11520216ec35a96bb1a57d0be6041b287c4528#diff-05e72230161a4d41c5a27fb453041c78839c193a9ddcb5066b1cb62c5a399529R129) I had to resort to - maybe this could be done more elegantly

Reply:
This has now been fixed